[GTK][WPE] Unreviewed test gardening

dpino opened the pull request:
#### 080cbab796309fe7f1d5f9654baf47b9b6dd5190
<pre>
[GTK][WPE] Unreviewed test gardening

Update test expectations for &apos;webrtc&apos; tests.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/296417@main">https://commits.webkit.org/296417@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1c34a584c145ed864d04fc51326d3f6b74927c3b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108456 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28117 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18541 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113665 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58859 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28806 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36667 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82360 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111404 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22846 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97688 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62796 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22262 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15825 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58385 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92217 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15879 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116786 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35510 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26159 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91384 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35883 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93962 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91185 "Found 2 new API test failures: /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebView/web-process-crashed, /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36075 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13842 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31255 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17517 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35412 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35122 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38473 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36803 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on LayoutTests/platform/glib/TestExpectations:
```diff
@@ -1442,6 +1440,104 @@ webkit.org/b/281343 media/media-source/media-managedmse-webvtt-track.html [ Skip
 #////////////////////////////////////////////////////////////////////////////////////////
 
+#////////////////////////////////////////////////////////////////////////////////////////
+# WebRTC-related bugs
```
There is already a section for WebRTC. Why add another one?

I don't understand the purpose of this PR :)

dpino replied:
The PR does several things. One thing is to update the test expectations of many tests. Some tests have been removed as well, as the have been passing consistently for a long time.

I didn't realised there was already a section for WebRTC. I can refactor the PR to avoid creating a new section. Right now, there are webrtc tests outside of this section, so I think it's still worth to put everything in a common WebRTC section. Hopefully after refactoring this PR the changes would be clearer.

philn commented on LayoutTests/platform/glib/TestExpectations:
```diff
@@ -2409,9 +2408,6 @@ webkit.org/b/235885 imported/w3c/web-platform-tests/webrtc/RTCRtpReceiver-getSyn
 webkit.org/b/235885 imported/w3c/web-platform-tests/webrtc/protocol/rtp-clockrate.html [ Skip ]
 webkit.org/b/235885 imported/w3c/web-platform-tests/webrtc-extensions/RTCRtpSynchronizationSource-senderCaptureTimeOffset.html [ Skip ]
 
-# We don't support spatial encoding yet.
-webkit.org/b/235885 webrtc/vp9-svc.html [ Skip ]
-
```
Are you sure this is passing now?

dpino replied:
Yes, this test is passing now.

If you have a recent build, you can try running this test:

```bash
Tools/Scripts/run-webkit-tests --no-show-results --no-new-test-results --no-sample-on-timeout 
--results-directory layout-test-results --debug-rwt-logging --no-retry-failures 
--wpe --release --force webrtc/vp9-svc.html
```

The test passes, although there's these lines in stderr: 

```bash
03:43:49.611 2 worker/0 webrtc/vp9-svc.html output stderr lines:
03:43:49.611 2   0:00:00.332042336    48 0x58ba5f261290 ERROR                default gstpipewirecore.c:91:make_core: (null)
03:43:49.611 2   0:00:00.332349001    48 0x58ba5f261290 ERROR               pipewire gstpipewiredeviceprovider.c:625:gst_pipewire_device_provider_start:<pipewiredeviceprovider0> Failed to connect
03:43:49.611 2   0:00:00.373610057    48 0x58ba5f261290 ERROR             video-info video-info.c:595:gst_video_info_from_caps: no format given
03:43:49.611 2   0:00:00.599271690    48 0x722698003bb0 ERROR               parsebin gstparsebin.c:3601:gst_parse_bin_expose:<parsebin1> Broken chain/group tree
03:43:49.611 2   0:00:00.686311693    48 0x58ba5f261290 ERROR             video-info video-info.c:595:gst_video_info_from_caps: no format given
03:43:49.611 2   0:00:00.858160522    48 0x722698006a10 ERROR               parsebin gstparsebin.c:3601:gst_parse_bin_expose:<parsebin3> Broken chain/group tree
```

philn replied:
Seems a bit odd, since GStreamer doesn't have SVC support for vp9... Oh well...

dpino replied:
Perhaps the test is not entirely correct.

philn commented on LayoutTests/platform/glib/TestExpectations:
```diff
@@ -2443,10 +2439,22 @@ imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-setLocalDescription-par
 # Creates 256 PeerConnections. The test takes around 30 seconds to complete.
 webrtc/datachannel/multiple-connections.html [ Slow ]
 
+webkit.org/b/276587 webrtc/video-av1.html [ Skip ]
```
Why is this skipped now?

dpino replied:
This test is timing out constantly according to results.webkit.org.

https://results.webkit.org/?platform=GTK&platform=WPE&suite=layout-tests&test=webrtc/video-av1.html&

philn replied:
That's a regression from [281394@main](https://commits.webkit.org/281394@main) apparently...

dpino committed:
[GTK][WPE] Unreviewed test gardening

Update test expectations for 'webrtc' tests.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:

Canonical link: https://commits.webkit.org/296417@main
